pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/ComplexTypeTransformer.java

```diff
@@ -340,7 +340,7 @@ private boolean containPrimitives(Collection value) {
   }
 
   protected static boolean isArray(Object obj) {
-    if (obj == null) {
```

Comment:
We should probably change it to `return obj instanceof Object[];`. Currently all the primitive arrays will fail